### PR TITLE
Move `production.Dockerfile` from slim to alpine

### PR DESCRIPTION
**Pull request by guidoiaquinti**
## Changes
1️⃣  move `production.Dockerfile` from the base image `python:3.8-slim` to `python:3.8-alpine3.14`.
2️⃣  remove the logic to disable SAML dependencies installation

## Why
1️⃣  See https://github.com/PostHog/posthog/pull/6524
2️⃣ See PR comments below

## How did you test this code?

1. The image is built correctly:

    ```
    ➜  posthog git:(alpine_prod_dockerfile) docker build -f production.Dockerfile .
    [+] Building 997.0s (18/18) FINISHED
    ```

2. Image size and build time are like before or improved

3. `docker-compose -f docker-compose.e2e.yml up` works:
    
    <details>
    
    ```
    posthog_server  | 🧐 Verifying installed packages...
    posthog_server  | [2021-10-25 09:10:12 +0000] [13] [INFO] Starting gunicorn 20.1.0
    posthog_server  |
    posthog_server  |
    posthog_server  |  _____          _   _    _
    posthog_server  | |  __ \        | | | |  | |
    posthog_server  | | |__) |__  ___| |_| |__| | ___   __ _
    posthog_server  | |  ___/ _ \/ __| __|  __  |/ _ \ / _` |
    posthog_server  | | |  | (_) \__ \ |_| |  | | (_) | (_| |
    posthog_server  | |_|   \___/|___/\__|_|  |_|\___/ \__, |
    posthog_server  |                                   __/ |
    posthog_server  |                                  |___/
    posthog_server  |
    posthog_server  |
    posthog_server  |
    posthog_server  | Server running on http://0.0.0.0:8000
    posthog_server  | Questions? Please shoot us an email at hey@posthog.com
    posthog_server  |
    posthog_server  | To stop, press CTRL + C
    posthog_server  | [2021-10-25 09:10:12 +0000] [13] [INFO] Listening at: http://0.0.0.0:8000 (13)
    posthog_server  | [2021-10-25 09:10:12 +0000] [13] [INFO] Using worker: gthread
    posthog_server  | [2021-10-25 09:10:12 +0000] [35] [INFO] Booting worker with pid: 35
    posthog_server  | [2021-10-25 09:10:12 +0000] [36] [INFO] Booting worker with pid: 36
    posthog_server  | yarn install v1.22.17
    posthog_server  |
    posthog_server  | 🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻
    posthog_server  | ️Environment variable DEBUG is set - PostHog is running in DEVELOPMENT MODE!
    posthog_server  | Be sure to unset DEBUG if this is supposed to be a PRODUCTION ENVIRONMENT!
    posthog_server  | 🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺
    posthog_server  |
    posthog_server  |
    posthog_server  | 🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻
    posthog_server  | ️Environment variable DEBUG is set - PostHog is running in DEVELOPMENT MODE!
    posthog_server  | Be sure to unset DEBUG if this is supposed to be a PRODUCTION ENVIRONMENT!
    posthog_server  | 🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺
    posthog_server  |
    posthog_server  | celery beat v4.4.2 (cliffs) is starting.
    posthog_server  |
    posthog_server  | 🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻
    posthog_server  | ️Environment variable DEBUG is set - PostHog is running in DEVELOPMENT MODE!
    posthog_server  | Be sure to unset DEBUG if this is supposed to be a PRODUCTION ENVIRONMENT!
    posthog_server  | 🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺
    posthog_server  |
    posthog_server  |
    posthog_server  | 🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻🔻
    posthog_server  | ️Environment variable DEBUG is set - PostHog is running in DEVELOPMENT MODE!
    posthog_server  | Be sure to unset DEBUG if this is supposed to be a PRODUCTION ENVIRONMENT!
    posthog_server  | 🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺🔺
    posthog_server  | [1/4] Resolving packages...
    posthog_server  |
    posthog_server  | [2/4] Fetching packages...
    posthog_server  | __    -    ... __   -        _
    posthog_server  | LocalTime -> 2021-10-25 09:10:14
    posthog_server  | Configuration ->
    posthog_server  |     . broker -> redis://redis:6379//
    posthog_server  |     . loader -> celery.loaders.app.AppLoader
    posthog_server  |     . scheduler -> redbeat.schedulers.RedBeatScheduler
    posthog_server  |        . redis -> redis://redis:6379/
    posthog_server  |        . lock -> `redbeat::lock` 45.00 seconds (45s)
    posthog_server  |     . logfile -> [stderr]@%WARNING
    posthog_server  |     . maxinterval -> 30.00 seconds (30s)
    posthog_server  |
    posthog_server  |  -------------- celery@63c6e20ab5b5 v4.4.2 (cliffs)
    posthog_server  | --- ***** -----
    posthog_server  | -- ******* ---- Linux-5.10.47-linuxkit-aarch64-with 2021-10-25 09:10:14
    posthog_server  | - *** --- * ---
    posthog_server  | - ** ---------- [config]
    posthog_server  | - ** ---------- .> app:         posthog:0xffffbc98baf0
    posthog_server  | - ** ---------- .> transport:   redis://redis:6379//
    posthog_server  | - ** ---------- .> results:     redis://redis:6379/
    posthog_server  | - *** --- * --- .> concurrency: 2 (prefork)
    posthog_server  | -- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
    posthog_server  | --- ***** -----
    posthog_server  |  -------------- [queues]
    posthog_server  |                 .> celery           exchange=celery(direct) key=celery
    posthog_server  |
    posthog_server  |
    posthog_server  | [2021-10-25 09:10:14,268: WARNING/MainProcess] /usr/local/lib/python3.8/site-packages/celery/fixups/django.py:202: UserWarning: Using settings.DEBUG leads to a memory
    posthog_server  |             leak, never use this setting in production environments!
    posthog_server  |   warnings.warn('''Using settings.DEBUG leads to a memory
    posthog_server  | [3/4] Linking dependencies...
    posthog_server  | warning "@posthog/plugin-server > snowflake-sdk > asn1.js-rfc2560@5.0.1" has unmet peer dependency "asn1.js@^5.0.0".
    posthog_server  | [4/4] Building fresh packages...
    posthog_server  | Done in 42.38s.
    posthog_server  | 🔁 Starting plugin server in a resiliency loop...
    posthog_server  | yarn run v1.22.17
    posthog_server  | $ posthog-plugin-server
    posthog_server  | [MAIN] 09:10:55 ⚡ @posthog/plugin-server v1.10.2
    posthog_server  | [MAIN] 09:10:55 ℹ️ 2 workers, 10 tasks per worker
    posthog_server  | [MAIN] 09:10:55 ✅ Connected to Redis!
    posthog_server  | [MAIN] 09:10:56 🍿 Fetched all actions from DB anew
    posthog_db      | 2021-10-25 09:10:56.049 UTC [60] ERROR:  relation "graphile_worker.migrations" does not exist at character 16
    posthog_db      | 2021-10-25 09:10:56.049 UTC [60] STATEMENT:  select id from "graphile_worker".migrations order by id desc limit 1;
    posthog_server  | [MAIN] 09:10:56 🚶 Connected to job queue producer "graphile"
    posthog_server  | [MAIN] 09:10:56 ⬇️ Fetching @posthog/mmdb for the first time
    posthog_server  | [MAIN] 09:10:56 ⏳ Downloading GeoLite2 database from PostHog servers...
    posthog_server  | [MAIN] 09:10:59 ✅ Downloaded GeoLite2-City-2021-10-19.mmdb of 24.5 MB
    posthog_server  | [MAIN] 09:10:59 💾 Saved GeoLite2-City-2021-10-19.mmdb into the database
    posthog_server  | [MAIN] 09:11:00 🪗 Decompressed GeoLite2-City-2021-10-19.mmdb from 24.5 MB into 71.8 MB
    posthog_server  | [MAIN] 09:11:00 🗺 Starting internal MMDB server...
    posthog_server  | [MAIN] 09:11:00 👂 Internal MMDB server listening on port 37065
    posthog_server  | [MAIN] 09:11:00 ⏰ Starting scheduling service...
    posthog_server  | [___2] 09:11:01 🧵 Starting Piscina worker thread 2…
    posthog_server  | [___1] 09:11:01 🧵 Starting Piscina worker thread 1…
    posthog_server  | [___1] 09:11:01 ✅ Connected to Redis!
    posthog_server  | [___2] 09:11:01 ✅ Connected to Redis!
    posthog_server  | [___2] 09:11:01 🍿 Fetched all actions from DB anew
    posthog_server  | [___1] 09:11:01 🍿 Fetched all actions from DB anew
    posthog_server  | [___2] 09:11:01 🚶 Connected to job queue producer "graphile"
    posthog_server  | [___1] 09:11:01 🚶 Connected to job queue producer "graphile"
    posthog_server  | [MAIN] 09:11:01 ⏰ Starting redlock "plugin-server:locks:schedule" ...
    posthog_server  | [MAIN] 09:11:01 ✅ Connected to Redis!
    posthog_server  | [MAIN] 09:11:01 🔄 Starting job queue consumer, trying to get lock...
    posthog_server  | [MAIN] 09:11:01 ⏰ Starting redlock "plugin-server:locks:job-queue-consumer" ...
    posthog_server  | [MAIN] 09:11:01 ✅ Connected to Redis!
    posthog_server  | [MAIN] 09:11:01 🍆 Starting Celery worker...
    posthog_server  | [MAIN] 09:11:01 ✅ Celery worker subscribed to posthog.tasks.plugins.plugin_job, posthog.tasks.process_event.process_event_with_plugins!
    posthog_server  | [MAIN] 09:11:01 🔒 Redlock "plugin-server:locks:schedule" acquired!
    posthog_server  | [MAIN] 09:11:01 🔒 Redlock "plugin-server:locks:job-queue-consumer" acquired!
    posthog_server  | [MAIN] 09:11:01 🔄 Job queue consumer lock acquired
    posthog_server  | [MAIN] 09:11:01 ✅ Connected to Redis!
    posthog_server  | [MAIN] 09:11:01 👀 Pub-sub started for channels: reload-plugins, reload-action, drop-action
    posthog_server  | [MAIN] 09:11:01 ✅ Connected to Redis!
    posthog_server  | [MAIN] 09:11:01 🚀 All systems go
    posthog_server  | [core] INFO: Worker connected and looking for jobs... (task names: 'pluginJob')
    ```
    </details>

4. Processes are ran and owned by the `posthog` user that is configured as expected:
    ``` 
    ~/code $ id posthog
    uid=101(posthog) gid=101(posthog) groups=101(posthog),101(posthog)

    ~/code $ echo $HOME
    /home/posthog

    ~/code $ ps aux | grep posthog
    1 posthog   0:00 {docker} /bin/bash ./bin/docker
    11 posthog   0:00 {docker-worker} /bin/bash ./bin/docker-worker
    12 posthog   0:00 {docker-server} /bin/bash ./bin/docker-server
    13 posthog   0:00 {plugin-server} /bin/bash ./bin/plugin-server
    14 posthog   0:00 {gunicorn} /usr/local/bin/python /usr/local/bin/gunicorn posthog.wsgi --config gunicorn.config.py --bind 0.0.0.0:8000 --log-file - --log-level info --access-logfile - --worker-tmp-dir /dev/shm --workers=2 --threads=4 --worker-class=gthread
    15 posthog   0:00 {docker-worker-c} /bin/bash ./bin/docker-worker-celery --with-scheduler
    16 posthog   0:00 {docker-worker-b} /bin/bash ./bin/docker-worker-beat
    17 posthog   0:00 {celery} /usr/local/bin/python /usr/local/bin/celery -A posthog worker -Ofair --without-gossip --without-mingle --without-heartbeat
    21 posthog   0:00 {celery} /usr/local/bin/python /usr/local/bin/celery -A posthog beat -S redbeat.RedBeatScheduler
    36 posthog   0:00 {gunicorn} /usr/local/bin/python /usr/local/bin/gunicorn posthog.wsgi --config gunicorn.config.py --bind 0.0.0.0:8000 --log-file - --log-level info --access-logfile - --worker-tmp-dir /dev/shm --workers=2 --threads=4 --worker-class=gthread
    37 posthog   0:00 {gunicorn} /usr/local/bin/python /usr/local/bin/gunicorn posthog.wsgi --config gunicorn.config.py --bind 0.0.0.0:8000 --log-file - --log-level info --access-logfile - --worker-tmp-dir /dev/shm --workers=2 --threads=4 --worker-class=gthread
    42 posthog   0:00 node /usr/local/bin/yarn start
    49 posthog   0:00 {celery} /usr/local/bin/python /usr/local/bin/celery -A posthog worker -Ofair --without-gossip --without-mingle --without-heartbeat
    54 posthog   0:00 {celery} /usr/local/bin/python /usr/local/bin/celery -A posthog worker -Ofair --without-gossip --without-mingle --without-heartbeat
    55 posthog   0:04 /usr/bin/node /home/posthog/code/plugins/node_modules/.bin/posthog-plugin-server
    68 posthog   0:00 /bin/sh
    77 posthog   0:00 ps aux
    78 posthog   0:00 grep posthog
    ```

### CODE REVIEW

**Comment by hazzadous** [comment]
@guidoiaquinti looks like the SAML build is failing.

This isn't directly relevant to this PR but I'd be interested to see how easy it would be to remove the bifurcation on SAML support. It seems it is just to resolve an issue with Heroku deploys, and seems a shame to complicate dev and ci just for this, see https://github.com/PostHog/posthog/pull/5870 for context.

**Comment by guidoiaquinti** [comment]
Anyone ✅ this?

**Comment by hazzadous** [comment]
> Anyone ✅ this?

I think everyones scared because no one wants to break production! What's the worst that can happen? Anything we need to monitor post merge/deploy?

**Comment by guidoiaquinti** [comment]
> I think everyones scared because no one wants to break production! What’s the worst that can happen? Anything we need to monitor post merge/deploy?

Let’s try to evaluate the risks so 🙌 

Unfortunately, we don’t have a proper dev/staging environment where to deploy this change safely after merge, so the options are few. @fuziontech @timgl any idea?

I think the worst thing it can happen is that take down production & we push a `latest` build on Docker Hub that is not working. The fix then should be something like:

1. we rollback the [ECS task definitions](https://console.aws.amazon.com/ecs/home?region=us-east-1#/taskDefinitions) for `migration`, `plugins`, `web` and `worker`. This will bring production back up.

1. in parallel, we revert this PR and we merge to `master`. Once the new image is built and deployed, everything should be back to normal.

I know ☝️  is not ideal but I don’t see a better way to test this with our current infra. Feedbacks are welcome!

File: production.Dockerfile [hunk 1]
```diff
@@ -1,69 +1,119 @@
...
-    && yarn --cwd plugins --frozen-lockfile --ignore-optional \
-    && yarn cache clean \
-    && rm -rf node_modules
+# Build the frontend
+#
+# Note: we run the build as a separate actions to increase
+# the cache hit ratio of the layers above.
+RUN yarn build && \
+    yarn cache clean && \
+    rm -rf ./node_modules ./plugins/node_modules
```
Comment (Twixes) [inline production.Dockerfile:102]: I may be missing something, but I don't think it's correct to delete `./plugins/node_modules`, as there's no `yarn build --cwd plugins` here?

Comment (hazzadous) [reply production.Dockerfile:102]: Plus we don't actually want to delete `./plugins/node_modules` at all as they are runtime deps for the plugin server?

We [install deps in `./bin/plugin-server` ](https://github.com/PostHog/posthog/blob/plugin-server-dates/bin/plugin-server#L43:L43)which is probably why it's not an issue but we probably shouldn't need to

**Comment by fuziontech** [comment]
>I think the worst thing it can happen is that take down production 

Naw, the health endpoint wouldn't come up and we'd just get stuck on an older deploy. Let's do this!
:shipit:

**Review by fuziontech — APPROVED** [review]
:shipit:

**Comment by fuziontech** [comment]
Oh, just remembered actually. This file is not even used for PostHog Cloud - check out this file: https://github.com/PostHog/posthog-cloud/blob/master/prod.web.Dockerfile

The build is triggered from here:
https://github.com/PostHog/posthog/blob/master/.github/workflows/prod-container.yml#L46

**Pull request merged**